src/module/guards/logged-in.guard.ts

```diff
@@ -0,0 +1,34 @@
+import { Inject, Injectable } from '@angular/core'
+import { CanActivate, Router } from '@angular/router'
+import { map } from 'rxjs/operators'
+import { Observable } from 'rxjs'
+
+import {
+  AuthProcessService,
+  NgxAuthFirebaseUIConfig,
+  NgxAuthFirebaseUIConfigToken,
+} from '../ngx-auth-firebase-u-i.module'
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoggedInGuard implements CanActivate {
+  constructor(
+    @Inject(NgxAuthFirebaseUIConfigToken)
+    private config: NgxAuthFirebaseUIConfig,
+    private router: Router,
+    private auth: AuthProcessService,
+  ) {}
+
+  canActivate(): Observable<boolean> {
+    return this.auth.afa.user.pipe(
```

Comment:
Do we need the user `Observable ` or the `authState` `Observable` ? 

I have in my project the following implementation of the `authGuard` 


```typescript
return new Promise((resolve, reject) => {
            this._angularFireAuth.authState.subscribe((user) => {
                if (user) {
                    user.getIdToken()
                        .then((token) => {
                            if (this.checkAccessToken(token)) { // the token can be expired ...
                                resolve(true);
                            }
                            else {
                                resolve(false);
                                // reject("access token invalid");
                            }
                        })
                   // ... other code
                }
```


```ts
private checkAccessToken(accessToken): boolean {
        let jwtHelper: JwtHelper = new JwtHelper();
        jwtHelper.decodeToken(accessToken); //only for debug
        jwtHelper.getTokenExpirationDate(accessToken); //Only for debug --> todo show exp date in the webapp
        return !jwtHelper.isTokenExpired(accessToken);
    }
``
`

which one do you think is more accurate ?

Reply:
the `AngularFireAuth.use` is actually the recommanded way to track and monitor the application's authentication state --> [info](https://github.com/angular/angularfire2/blob/master/docs/auth/getting-started.md) 👍